### PR TITLE
8292628: x86: Improve handling of constants in trigonometric stubs

### PR DESCRIPTION
**Pull request by iwanowww**
Improve how constants are handled in trigonometric stubs on x86. 
In particular, extract shared constants into a separate file and 
encapsulate them in MacroAssembler.

Also, some minor cleanups included. 

Testing:
 - [x] hs-tier1 - hs-tier4
 - [x] x86_32 build

/cc hotspot

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292628](https://bugs.openjdk.org/browse/JDK-8292628): x86: Improve handling of constants in trigonometric stubs

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to [98bfa9ec](https://git.openjdk.org/jdk/pull/9923/files/98bfa9eca9c173ad14d652e64def63ac6dcdc434)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9923/head:pull/9923` \
`$ git checkout pull/9923`

Update a local copy of the PR: \
`$ git checkout pull/9923` \
`$ git pull https://git.openjdk.org/jdk pull/9923/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9923`

View PR using the GUI difftool: \
`$ git pr show -t 9923`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9923.diff">https://git.openjdk.org/jdk/pull/9923.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/x86/macroAssembler_x86_constants.cpp [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+};
+address MacroAssembler::ONES = (address)_ONES;
+
+ATTRIBUTE_ALIGNED(16) juint _PI4_INV[] = {
+    0x6dc9c883UL, 0x3ff45f30UL
+};
+address MacroAssembler::PI4_INV = (address)_PI4_INV;
+
+ATTRIBUTE_ALIGNED(16) juint _PI4X3[] = {
+    0x54443000UL, 0xbfe921fbUL, 0x3b39a000UL, 0x3d373dcbUL, 0xe0e68948UL, 0xba845c06UL
```
Comment (shipilev) [inline src/hotspot/cpu/x86/macroAssembler_x86_constants.cpp:251]: While at it, maybe we should stick to one style for these tables. The large table above has five columns, so maybe wrap this and following table like that too?

File: src/hotspot/cpu/x86/macroAssembler_x86_exp.cpp [hunk 2]
```diff
@@ -191,32 +177,31 @@ ATTRIBUTE_ALIGNED(4) juint _ONE_val[] =
...
+  address INF      = (address)_INF;
+
   subq(rsp, 24);
   movsd(Address(rsp, 8), xmm0);
   unpcklpd(xmm0, xmm0);
   movdqu(xmm1, ExternalAddress(cv));       // 0x652b82feUL, 0x40571547UL, 0x652b82feUL, 0x40571547UL
-  movdqu(xmm6, ExternalAddress(Shifter));  // 0x00000000UL, 0x43380000UL, 0x00000000UL, 0x43380000UL
+  movdqu(xmm6, ExternalAddress(SHIFTER));  // 0x00000000UL, 0x43380000UL, 0x00000000UL, 0x43380000UL
   movdqu(xmm2, ExternalAddress(16 + cv));    // 0xfefa0000UL, 0x3f862e42UL, 0xfefa0000UL, 0x3f862e42UL
   movdqu(xmm3, ExternalAddress(32 + cv));    // 0xbc9e3b3aUL, 0x3d1cf79aUL, 0xbc9e3b3aUL, 0x3d1cf79aUL
```
Comment (shipilev) [inline src/hotspot/cpu/x86/macroAssembler_x86_exp.cpp:206]: If you are flipping `+` operands near the changes, maybe flip these too?

**Review by shipilev — APPROVED** [review]
Looks okay with some nits.

File: src/hotspot/cpu/x86/macroAssembler_x86_constants.cpp (hunk 1 above)
Comment (iwanowww) [reply src/hotspot/cpu/x86/macroAssembler_x86_constants.cpp]: Sounds good. Fixed.

File: src/hotspot/cpu/x86/macroAssembler_x86_exp.cpp (hunk 2 above)
Comment (iwanowww) [reply src/hotspot/cpu/x86/macroAssembler_x86_exp.cpp]: Fixed.

**Review by vnkozlov — APPROVED** [review]
Seems fine to me. Thank you for doing this cleanup.

**Comment by iwanowww** [comment]
Thanks for the reviews, Aleksey and Vladimir.

/integrate

**Pull request closed without merging**